blaze-client/src/main/scala/org/http4s/blaze/client/BlazeClient.scala

```diff
@@ -85,6 +87,9 @@ object BlazeClient {
           borrow.use { next =>
             val res: F[Resource[F, Response[F]]] = next.connection
               .runRequest(req)
+              .adaptError { case EOF =>
+                new SocketException(s"HTTP connection closed: ${key}")
```

Comment:
It's unusual in the JVM to log the host info with the `IOException`s, which has me wondering if there's a reason for that.  But it's consistent with what blaze does on connection failures.